src/1_utils.js

```diff
@@ -750,6 +750,62 @@ utils.removePropertiesFromObject = function(objectToModify, keysToRemove) {
 	}
 };
 
+// v2/event utility functions
+
+var BRANCH_STANDARD_EVENTS = [ 'ADD_TO_CART', 'ADD_TO_WISHLIST', 'VIEW_CART', 'INITIATE_PURCHASE', 'ADD_PAYMENT_INFO', 'PURCHASE', 'SPEND_CREDITS', 'SEARCH', 'VIEW_ITEM', 'VIEW_ITEMS', 'RATE', 'SHARE', 'COMPLETE_REGISTRATION', 'COMPLETE_TUTORIAL', 'ACHIEVE_LEVEL', 'UNLOCK_ACHIEVEMENT' ];
+var BRANCH_STANDARD_EVENT_DATA = [ 'transaction_id', 'revenue', 'currency', 'shipping', 'tax', 'coupon', 'affiliation', 'search_query', 'description' ];
+
+utils.isStandardEvent = function(eventName) {
+	return BRANCH_STANDARD_EVENTS.indexOf(eventName) > -1;
+};
+
+utils.separateEventAndCustomData = function(eventAndCustomData) {
+
+	if (!eventAndCustomData || Object.keys(eventAndCustomData).length === 0) {
+		return null;
+	}
+	var customDataKeys = utils.calculateDiffBetweenArrays(BRANCH_STANDARD_EVENT_DATA, Object.keys(eventAndCustomData));
+	var customData = {};
+
+	for (var i = 0; i < customDataKeys.length; i++) {
+		var key = customDataKeys[i];
+		customData[key] = eventAndCustomData[key];
+		delete eventAndCustomData[key];
+	}
+
+	return {
+		"custom_data": customData,
+		"event_data": eventAndCustomData
+	};
+};
+
+utils.validateParameterType = function(parameter, type) {
+	if (!parameter || !type) {
+		return false;
+	}
+	if (type === "array") {
+		return Array.isArray(parameter);
+	}
+	return typeof parameter === type && !Array.isArray(parameter);
+};
+
+// Used by logEvent() to send fields related to user's visit and device to v2/event standard and custom
+// Requires a reference to the branch object to access information such as browser_fingerprint_id
+utils.getUserData = function(branch) {
+	var user_data = {};
+	user_data = utils.addPropertyIfNotNull(user_data, "http_origin", document.URL);
+	user_data = utils.addPropertyIfNotNull(user_data, "user_agent", navigator.userAgent);
+	user_data = utils.addPropertyIfNotNull(user_data, "language", navigator.language);
+	user_data = utils.addPropertyIfNotNull(user_data, "screen_width", screen.width);
+	user_data = utils.addPropertyIfNotNull(user_data, "screen_height", screen.height);
+	user_data = utils.addPropertyIfNotNull(user_data, "http_referrer", document.referrer);
+	user_data = utils.addPropertyIfNotNull(user_data, "identity_id", branch.identity_id);
```

Comment:
i don't believe we're collecting identity id, we do collect developer_identity though.